modules/custom/az_cas/src/Form/AzCasSettingsForm.php

```diff
@@ -71,34 +72,89 @@ public function getFormId() {
       '#default_value' => $az_cas_config->get('disable_login_form'),
     ];
 
-    $form['disable_admin_add_user_button'] = [
-      '#title' => t("Disable 'Add user' button"),
+    $form['login_settings']['disable_password_recovery_link'] = [
+      '#title' => $this->t("Disable 'request new password' form"),
+      '#type' => 'checkbox',
+      '#description' => $this->t("Disables the default password recovery functionality provided by Drupal core."),
+      '#default_value' => $az_cas_config->get('disable_password_recovery_link'),
+    ];
+
+    $form['admin_settings'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Admin Settings'),
+      '#description' => $this->t('Configure how CAS interacts with the admin interface.'),
+    ];
+
+    $form['admin_settings']['disable_admin_add_user_button'] = [
+      '#title' => $this->t("Disable 'Add user' button"),
       '#type' => 'checkbox',
-      '#description' => t("Removes button for adding non-CAS users in admin interface."),
+      '#description' => $this->t("Removes button for adding non-CAS users in admin interface."),
       '#default_value' => $az_cas_config->get('disable_admin_add_user_button'),
     ];
 
-    $form['disable_password_recovery_link'] = [
-      '#title' => t("Disable 'request new password' form"),
+    $form['guest_authentication'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Guest Authentication'),
+    ];
+
+    $form['guest_authentication']['guest_mode'] = [
       '#type' => 'checkbox',
-      '#description' => t("Disables the default password recovery functionality provided by Drupal core."),
-      '#default_value' => $az_cas_config->get('disable_password_recovery_link'),
+      '#title' => $this->t('Enable guest authentication mode'),
+      '#description' => $this->t('Allows restricting access to certain paths to anyone with a valid NetID without creating Drupal user accounts.'),
+      '#default_value' => $az_cas_config->get('guest_mode'),
+    ];
+
+    $form['guest_authentication']['guest_auth_settings'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Guest Authentication Settings'),
+      '#open' => TRUE,
+      '#states' => [
+        'visible' => [
+          ':input[name="guest_mode"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    // Add notice about auto_register requirement.
+    $cas_settings_url = Url::fromRoute('cas.settings')->toString() . '#edit-user-accounts';
+    $form['guest_authentication']['guest_auth_settings']['auto_register_notice'] = [
+      '#type' => 'markup',
+      '#markup' => '<div class="messages messages--warning">' . $this->t('Guest authentication requires the CAS module\'s "Auto register users" setting to be enabled. This setting will be automatically enabled when guest mode is activated. <a href="@cas_settings_url">View CAS settings</a>', ['@cas_settings_url' => $cas_settings_url]) . '</div>',
```

Comment:
We should add a reminder here that people should not be entering protected data even though this is "password protected".

Reply:
We might also add some advice about the kinds of paths to add here, for example a canonical path like /node/33 could be considered better than a path alias like `/pages/photo-galleries`  because it is the permanent address unless deleted or unpublished of course.